plugin/pkg/shoot/validator/admission.go

```diff
@@ -687,15 +687,16 @@ func (c *validationContext) validateKubernetes(a admission.Attributes) field.Err
 		return nil
 	}
 
-	ok, isDefaulted, validKubernetesVersions, versionDefault := validateKubernetesVersionConstraints(c.cloudProfile.Spec.Kubernetes.Versions, c.shoot.Spec.Kubernetes.Version, c.oldShoot.Spec.Kubernetes.Version)
-	if !ok {
-		err := field.NotSupported(path.Child("version"), c.shoot.Spec.Kubernetes.Version, validKubernetesVersions)
-		if isDefaulted {
-			err.Detail = fmt.Sprintf("unable to default version - couldn't find a suitable patch version for %s. Suitable patch versions have a non-expired expiration date and are no 'preview' versions. 'Preview'-classified versions have to be selected explicitly -  %s", c.shoot.Spec.Kubernetes.Version, err.Detail)
-		}
-		allErrs = append(allErrs, err)
-	} else if versionDefault != nil {
-		c.shoot.Spec.Kubernetes.Version = versionDefault.String()
+	defaultVersion, errList := defaultKubernetesVersion(c.cloudProfile.Spec.Kubernetes.Versions, c.shoot.Spec.Kubernetes.Version, path.Child("version"))
+	if len(errList) > 0 {
+		allErrs = append(allErrs, errList...)
+	}
+
+	if defaultVersion != nil {
+		c.shoot.Spec.Kubernetes.Version = *defaultVersion
+	} else {
+		// We assume that the 'defaultVersion' is already calculated correctly, so only run validation if the verion was not defaulted.
```

Comment:
```suggestion
		// We assume that the 'defaultVersion' is already calculated correctly, so only run validation if the version was not defaulted.
```

Reply:
Please see eb3963710042358feabb98121d3c2833928bb9d2 😓 
I also tested this change WRT backwards compatibility and Protobuf: An "older" Gardenlet version was still able to connect to a "new" Apiserver ✅